scalableminds/webknossos
# Measure used disk storage

webKnossos now measures the storage used by the datasets of the organizations.
New config options `datastore.reportUsedStorage.enabled` and `webKnossos.fetchUsedStorage.interval`.
Note that the organization is re-scanned regularly, but dataset upload + delete trigger a refresh of said dataset directly.
Datastores report that they have reportUsedStorage.enabled to webKnossos, which then stores that info in the postgres datastores table.
Load distribution strategy assumes that measuring a single organization isn’t that huge an operation (optimized for multi-orga setup).

### Steps to test:
- turn on storage reporting as described above
- start up, after initial delay, the organization api request should include used storage
- test with multiple organizations that the load distribution logic works

### Issues:
- fixes #6614 

------
